src/main/java/net/fabricmc/loader/entrypoint/minecraft/hooks/EntrypointClient.java

```diff
@@ -30,6 +31,7 @@ public static void start(File runDir, Object gameInstance) {
 
 		FabricLoader.INSTANCE.prepareModInit(runDir, gameInstance);
 		EntrypointUtils.invoke("main", ModInitializer.class, ModInitializer::onInitialize);
+		FabricLoader.INSTANCE.getEntrypoints("fake_mods", FakeModProvider.class).forEach(provider -> FabricLoader.INSTANCE.appendFakeMods(provider.getFakeMods()));
```

Comment:
The question I have is, is this the right spot for the FakeModProviders to run?

I personally think that we should load fake mods after the GameProvider is set but before onInitialize after each of the ModInitializers fires their constructors. 
One key reason for doing this before Client/Common/Server entrypoints is a concern about mod loading order, where if you do this after the Common Initializer has gone off you don't exactly have much context in what all mods that exist at that point along with breaking the expectation of the ModInitalizers having a final state for mods being loaded.

Reply:
The main worry there is that then the fake mods likely haven't been loaded. You'd almost definitely want to do that in `onInitialize`.

Reply:
What is your reason for loading fake mods after the mod initializer (thereby the client/server would be loaded). Sponge's events cover some states near what on Initialize would cover.